packages/subgraph/schema.graphql

```diff
@@ -220,6 +197,25 @@ type Dao @entity {
   plugins: [PluginInstallation!]! @derivedFrom(field: "dao")
 }
 
+# ACL
+
+# Cannot be immutable because a permission can be revoked and granted in the same TX.
+# This results in an error in the subgraph if the entity is immutable.
+# No need to store granted as we can delete permission when revoked
+type Permission @entity {
+  id: ID! # emitting contract Address (DAO or PluginRepo) + where + permissionId + who
+  where: Bytes!
+  permissionId: Bytes!
+  who: Bytes!
+  actor: Bytes!
+  condition: Bytes
+
+  # The following attributes is used to create
+  # one-to-many between Dao or PluginRepo to Permission
+  dao: Dao
+  pluginRepo: PluginRepo
```

Comment:
I understand that these are the contracts that currently inherit from `PermissionManager`. 
Why do we need to discriminate between DAO and PluginRepo?
Why can't we just have one member `permissionManager` that holds the address to either one of them?

Reply:
Reading https://thegraph.com/docs/en/developing/creating-a-subgraph/, cannot we have a `PermissionManager` entity that both PluginRepo and DAO inherit from?

Maybe a naive question. I need to become better with subgraph

Reply:
yes we can create the type `PermissionManager {id, dao, pluginRepo}`  but then u have to repeat the same `dao`, `pluginRepo`
type union is now allowed by subgraph, even it is allowed by graphQl
so this is the most economic way i thought of

Reply:
Ok, just wanted to confirm that this is due to subgraph limitaitons.